extensions/composer/extension.py

```diff
@@ -303,8 +290,10 @@ def _build_composer_environment(self):
             env[key] = val if type(val) == str else json.dumps(val)
 
         # add basic composer vars
-        env['COMPOSER_VENDOR_DIR'] = self._ctx['COMPOSER_VENDOR_DIR']
-        env['COMPOSER_BIN_DIR'] = self._ctx['COMPOSER_BIN_DIR']
+        if not self._ctx['COMPOSER_VENDOR_DIR']:
+            env['COMPOSER_VENDOR_DIR'] = self._ctx['COMPOSER_VENDOR_DIR']
+        if not self._ctx['COMPOSER_BIN_DIR']:
+            env['COMPOSER_BIN_DIR'] = self._ctx['COMPOSER_BIN_DIR']
```

Comment:
Curious by you're setting not COMPOSER_BIN_DIR.  The reason that we set this normally is so that any binaries / scripts installed by Composer are on the PATH and you can execute them.  Do you feel like this is problematic?